### PR TITLE
[ci] adding github action (closes #156)

### LINKED ISSUE #156: Consider move to GH Actions
Will allow leveraging useful actions, such as https://github.com/treebeardtech/treebeard

### PR DESCRIPTION
**Pull request by djarecka**
@mgxd, @effigies @satra - testing github actions, happy to get any feedback (especially if you have better idea how to deal with container management/caching)

and I'd really want to know how to enable it... when I was testing on my profile I haven't had to do anything extra, it just worked when i included .github/workflows, but here it doesn't (at least for this pr)

this is a ga dashboard from my fork (tests are failing right now):
https://github.com/djarecka/nipype_tutorial/runs/957128296?check_suite_focus=true

### CODE REVIEW

**Comment by satra** [comment]
this is still testing via circle. actions only start executing after they have at least been added to master. so if it works for you, merge this in and then the action will fire. after that PRs should do the right thing.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      run: docker build . --file Dockerfile -t nipype_tutorial:latest
+
+  test_1:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: satackey/action-docker-layer-caching@v0.0.5
+    - name: run test 1
+      run: docker run -it --rm nipype_tutorial:latest python /home/neuro/nipype_tutorial/test_notebooks.py 1
```
Comment (mgxd) [inline .github/workflows/testing.yml:31]:
```suggestion
      run: docker run --rm nipype_tutorial:latest python /home/neuro/nipype_tutorial/test_notebooks.py 1
```

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  test_2:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: satackey/action-docker-layer-caching@v0.0.5
+    - name: run	test 2
+      run: docker run -it --rm nipype_tutorial:latest python /home/neuro/nipype_tutorial/test_notebooks.py 2
```
Comment (mgxd) [inline .github/workflows/testing.yml:40]:
```suggestion
      run: docker run --rm nipype_tutorial:latest python /home/neuro/nipype_tutorial/test_notebooks.py 2
```

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  test_3:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: satackey/action-docker-layer-caching@v0.0.5
+    - name: run	test 3
+      run: docker run -it --rm nipype_tutorial:latest python /home/neuro/nipype_tutorial/test_notebooks.py 3
```
Comment (mgxd) [inline .github/workflows/testing.yml:49]:
```suggestion
      run: docker run --rm nipype_tutorial:latest python /home/neuro/nipype_tutorial/test_notebooks.py 3
```

File: Dockerfile [hunk 4]
```diff
@@ -9,6 +10,8 @@
 
 FROM neurodebian:stretch-non-free
 
+USER root
```
Comment (mgxd) [inline Dockerfile:13]: isn't the user originally root by default?

File: .github/workflows/testing.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
```
Comment (mgxd) [inline .github/workflows/testing.yml:8]:
```suggestion
  schedule:
    - cron: "0 10 * * *"

```

**Comment by djarecka** [comment]
a fun fact - the current version of the `test_notebooks` (2 years old?) where giving green light on CI regardless if tests were passing or failing... 
see for example `run_tests` section of circleCI: https://app.circleci.com/pipelines/github/miykael/nipype_tutorial/87/workflows/cbc2e7db-0dd5-4352-bd9c-fb767a6b7975/jobs/1352

**Comment by mgxd** [comment]
@djarecka it looks like the problem lies within the test script:
https://github.com/miykael/nipype_tutorial/blob/ff7c4232aa3371254d444271e7058c7d620c66f6/test_notebooks.py#L94-L96

since the return value of the system call is not checked, any command would pass.

**Comment by djarecka** [comment]
@mgxd - yes, I noticed... Will fix this late, and than would have to review the notebooks that are failing, don't even want to know when  they started failing....

**Comment by djarecka** [comment]
actually, I might just merge this first to have at least working CI structure (i will give up on finding out why CircleCI doesn't want to build the image for me)

**Comment by djarecka** [comment]
@satra - you were right, it works now! :) 
in my fork i didn't need to merge to master, but perhaps it makes sense to treat owners of the repo differently

**Pull request merged**